src/oom/oomd-manager.c

```diff
@@ -404,21 +406,35 @@ static int monitor_swap_contexts_handler(sd_event_source *s, uint64_t usec, void
                         log_debug_errno(r, "Failed to get monitored swap cgroup candidates, ignoring: %m");
 
                 threshold = m->system_context.swap_total * THRESHOLD_SWAP_USED_PERCENT / 100;
-                r = oomd_kill_by_swap_usage(candidates, threshold, m->dry_run, &selected);
-                if (r == -ENOMEM)
-                        return log_oom();
-                if (r < 0)
-                        log_notice_errno(r, "Failed to kill any cgroup(s) based on swap: %m");
+
+                n = oomd_sort_cgroup_contexts(m->monitored_swap_cgroup_contexts,
+                                              compare_cgroup_path_and_swap_usage,
```

Comment:
Yeah, I see what you mean. How exactly should we determine when a swap kill candidate's OOM preference should be honored then? Since swap kill candidates are sorted all at once, regardless of which monitored cgroup context they are descendant of, we cannot use the same logic as the memory pressure scenario (this commit was an attempt to make that possible).

If we just dropped "oomd: calculate swap kill candidates one monitored cgroup at a time" from this PR, then the existing behavior (only cgroups owned by root can set `ManagedOOMPreference`) would remain for the swap kill scenario.

Reply:
> If we just dropped "oomd: calculate swap kill candidates one monitored cgroup at a time" from this PR, then the existing behavior (only cgroups owned by root can set ManagedOOMPreference) would remain for the swap kill scenario.

I think this makes the most sense. Given that swap kill thresholds are globally controlled, root should be able to use `ManagedOOMPreference=`. Non-root cgroups that don't want to get killed could have someone privileged set up a more fine grained policy where certain cgroups are omitted altogether from the candidate list. For example:
```
/ [ManagedOOMSwap=auto]
  /system.slice [ManagedOOMSwap=kill]
  /user.slice [ManagedOOMSwap=auto]
    /user.slice/user-1000.slice [ManagedOOMSwap=kill]
    /user.slice/user-3000.slice [ManagedOOMSwap=auto]
```
Which would skip `/user.slice/user-3000.slice` as a swap kill candidate, but still include everything else

Reply:
Okay, I have removed this commit from the PR. I believe the test failures are unrelated.